apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOptimize/Components/AllowChar.tsx

```diff
@@ -159,23 +170,46 @@ export default function AllowChar({
       characterRarityTotals: [...allCharacterRarityKeys],
       locListTotals: ['allowed', 'excluded'],
     } as const
+    let travelerProcessed = false
+
     return bulkCatTotal(catKeys, (ctMap) =>
       Object.entries(database.chars.data)
-        .filter(([ck]) => ck !== characterKey)
+        .filter(
+          ([ck]) =>
+            charKeyToLocCharKey(ck) !== charKeyToLocCharKey(characterKey)
+        )
         .forEach(([ck]) => {
           const sheet = getCharSheet(ck, database.gender)
+
           const eleKey = sheet.elementKey
+          const weaponTypeKey = sheet.weaponTypeKey
+          const characterRarityKey = sheet.rarity
+
           ctMap.elementTotals[eleKey].total++
           if (charKeyMap[ck]) ctMap.elementTotals[eleKey].current++
 
-          const weaponTypeKey = sheet.weaponTypeKey
+          if (charKeyToLocCharKey(ck) === 'Traveler') {
+            console.log(charKeyMap)
+            if (!charKeyMap[ck] && locList.includes('Traveler')) {
```

Comment:
Please add some comments to why we do this logic

Reply:
Added explanations, hopefully they are understandable